tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -99,9 +122,16 @@ class AppFilterImpl(
     } else {
       appNameFiltered
     }
+
     val appTimeFiltered = if (appArgs.startAppTime.isSupplied) {
       val msTimeToFilter = AppFilterImpl.parseAppTimePeriodArgs(appArgs)
-      val filtered = userNameFiltered.filter { app =>
+      val logicFiltered = if (appArgs.any()) {
+        apps
+      } else {
+        userNameFiltered
+      }
+
+      val filtered = logicFiltered.filter { app =>
```

Comment:
This can be simplified by replacing

```scala
val filtered = logicFiltered.filter { app =>
  val appStartOpt = app.appInfo.map(_.startTime)
  if (appStartOpt.isDefined) {
    appStartOpt.get >= msTimeToFilter
  } else {
    false
  }
}
filtered
```

with

```scala
logicFiltered.filter(_.appInfo.exists(_.startTime >= msTimeToFilter))
```

Reply:
Thanks! This is easy to read.